test/autodiff_tests.jl

```diff
@@ -1,29 +1,32 @@
 import FiniteDiff
 import ForwardDiff
 import ReverseDiff
-# import Zygote
+import Zygote
 
 using Test
 
-grads = let 
-    F(x, θ, deg) = (θ[1] - x[1])^deg + θ[2] * (x[2] - x[1]^deg)^deg
+F(x, θ, deg) = (θ[1] - x[1])^deg + θ[2] * (x[2] - x[1]^deg)^deg
+F_idx_val(ca) = F(ca[Val(:x)], ca[Val(:θ)], ca[Val(:deg)])
+F_idx_sym(ca) = F(ca[:x], ca[:θ], ca[:deg])
+F_prop(ca) = F(ca.x, ca.θ, ca.deg)
 
-    F_idx_val(ca) = F(ca[Val(:x)], ca[Val(:θ)], ca[Val(:deg)])
-    F_idx_sym(ca) = F(ca[:x], ca[:θ], ca[:deg])
-    F_prop(ca) = F(ca.x, ca.θ, ca.deg)
+ca = ComponentArray(x = [1, 2], θ = [1.0, 100.0], deg = 2)
+truth = [-400, 200]
 
-    ca = ComponentArray(x = [1, 2], θ = [1.0, 100.0], deg = 2)
+@testset "$(nameof(F_))" for F_ in (F_idx_val, F_idx_sym, F_prop)
+    finite = FiniteDiff.finite_difference_gradient(ca -> F_(ca), ca).x
+    @test finite ≈ truth
 
-    (
-        truth = [-400, 200],
-        finite = FiniteDiff.finite_difference_gradient(ca -> F_prop(ca), ca).x,
-        forward = ForwardDiff.gradient(ca -> F_prop(ca), ca).x,
-        reverse = ReverseDiff.gradient(ca -> F_prop(ca), ca).x,
-        # zygote = Zygote.gradient(ca -> F(ca), ca)[1].x,
-    )
-end
+    forward = ForwardDiff.gradient(ca -> F_(ca), ca).x
+    @test forward ≈ truth
+
+    reverse = ReverseDiff.gradient(ca -> F_(ca), ca).x
+    if F_ in (F_idx_val, F_idx_sym)
+        @test_broken reverse ≈ truth
```

Comment:
Produces a zero-vector rather than `truth`. Guessing there's something wrong with `src/compat/reversediff.jl` but I haven't investigated this further.